src/yahtzee_clj/core.clj

```diff
@@ -9,11 +10,80 @@
   "Rolls a hand of 5 six sided dice."
   (repeatedly 5 roll))
 
-(defn score-singles [hand]
-  "Scores the singles."
+(defn score-chance [hand]
+  "Scores the chance - i.e. sum of hand."
   (reduce + hand))
 
+(defn score-singles [hand single]
+  "Scores the singles"
+  (reduce + (filter #(= single %) hand))
+  )
+
+(defn score-of-a-kind [hand kind-type]
+  "Scores three, four and five of a kind"
+  (let [most-common (last (sort-by val (frequencies hand)))
+        face (first most-common)
+        count (last most-common)]
+    (cond
+      (< count 3) 0
+      (and (>= count 3) (= kind-type 3)) (* face 3)
+      (and (>= count 4) (= kind-type 4)) (* face 4)
+      (and (= count 5) (= kind-type 5)) 50
```

Comment:
I don't think these are the rules of yahtzee! If you get 3 dice the same you qualify. The score is the sum of all dice.